### ISSUE TITLE
Add support for negative `OFFSET`

### ISSUE THREAD

**Issue by vpetrovykh** [edgedb/edgedb#211]
It would be nice to support negative `OFFSET` values. The interpretation would be similar to Python, where a negative index indicates counting from the end.

In terms of implementation negative `OFFSET` is equivalent to an order reversal, `LIMIT` and another order reversal:

```
SELECT Foo
ORDER BY Foo.bar ASC NULLS LAST
OFFSET -3;

# equivalent to
SELECT (
    SELECT Foo
    ORDER BY Foo.bar DESC NULLS FIRST
    LIMIT 3
  )
ORDER BY Foo.bar ASC NULLS LAST;
```

**Comment by vpetrovykh** [comment]
This feature would also be used in GraphQL to implement `last` argument for pagination.

**Comment by msullivan** [comment]
How would we handle this in the general case, where the argument to `OFFSET` is dynamic? That is probably actually the typical case, since the constant will be replaced with a query parameter.

I guess the question is less "how", since I think it is definitely doable (maybe: order by row_number()*sign(offset), then offset by abs(offset), then order by row_number), and more "is it worth doing that"?

**Comment by elprans** [comment]
`OFFSET`/`LIMIT` without `ORDER BY` is almost always meaningless anyway, so one possibility is that we require `ORDER BY` for negatives?

**Comment by msullivan** [comment]
I don't think that really addresses my concern, which is that the argument to OFFSET is dynamic and we don't know whether it is negative or not, and so need to generate code that handles both cases.

**Comment by elprans** [comment]
> we don't know whether it is negative or not, and so need to generate code that handles both cases.

Ah, that's a good point.  Perhaps negative slicing should be explicit syntactically?  E.g `LIMIT NEGATIVE -10` or something?
